lib/internal/linkedlist.js

```diff
@@ -42,10 +49,17 @@ exports.remove = remove;
 
 // remove a item from its list and place at the end.
 function append(list, item) {
-  remove(item);
+  if (item._idleNext || item._idlePrev) {
```

Comment:
linkedlist.js implements a circular list, the expect case is that both will be set or not set together.
This is just a fast-path optimization for newly created list nodes that have never been linked.